### ISSUE TITLE
Multisend Issue with Not Signing Second TX - v0.9.23

### ISSUE THREAD

**Issue by davestaxcp** [jdogresorg/freewallet-desktop#152]
Some users are experiencing an issue where a Multisend fails on the second Signing of Transaction in v0.9.23
These sends do not include subassets (if that matters)

Some solutions I have seen that may help is:

- Changing the BTC fee (something custom but near the wanted fee)
- Sending less assets in the multisend (10 or less)
- Refreshing Freewallet (to the latest BTC block if left open for some time) and trying again
- Restarting Freewallet and trying again

Some other issues that may be related to this one if I was to start theorizing ...

- Users have reported an error trying to use multi send with Trezor hardware wallets and in one circumstance had an issue using the multisend function while having a separate hardware address imported into Freewallet. May not be related but thought I would include the data.

Another thing to note here is less users have reported multisend issues with the prior version (0.9.22) regarding the "Not Signing Second TX" issue 

So it may also help to downgrade to 0.9.22 for a very large single multisend

**Comment by davestaxcp** [comment]
@oskgie on Telegram mentioned: have unadded the trezor addresses like you suggested... and everything works now. just sent >100 cards from my normal address via multisend successfully.

Wow is the trezor hardware addresses import related here??

**Comment by B0BSmiths** [comment]
I can confirm multisend does not work with Trezor.  The second part of the p2sh is not signed correctly.  As the Trezor device does not speak Counterparty it is not aware of the signatures required on the encoded inputs

**Comment by subterranean1** [comment]
Yesterday (1/24/24) using Version 0.9.25 AND 0.9.26 I got an error when attempting to do a multi-send. FW would broadcast the first transaction but then would send an error message during transaction 2. BTC got sent to a random address each time which I presume is lost forever. I attempted the multi-send 4 times. Twice on a laptop with no hardware wallet addresses and twice on a Windows desktop that DOES have hardware addresses imported. 

I did not capture all the errors, but here are two: 

Error validating transaction: Error running script for input 0 referencing bd58ff9e65feb25f57033b4ef5ddaf58e5189c033996edb591cac234fec55af7 at 0: scriptsig not push only for P2SH.

Error validating transaction: Rejected script for input 0 referencing dd66b8bce0ddc86a9a4cce7603ab8d78a57ce48c1bfa2dcf5d8767903fd62632 at 0..

**Comment by jdogresorg** [comment]
> I can confirm multisend does not work with Trezor. 

@B0BSmiths Correct, I will be adding MPMA support for trezor in the future... just need to update the code a bit so that we handle signing the second tx... Javier and I will dig into this when we have some time.

> Yesterday (1/24/24) using Version 0.9.25 AND 0.9.26 I got an error when attempting to do a multi-send.

@subterranean1 FYI, the outputs in MPMA tx1 are still signable/spendable... not random... so, even if you generate part 1 of a 2 part MPMA tx, then part 2 fails... you are still able to gather/collect all the dust from those part1 txs.

Currently no tool exists to find these failed half-way-done MPMA sends and gather the dust up, but its technically possible, and i'm sure tools will be built in the future to provide the ability to gather dust with a few clicks.

TLDR, your not losing any BTC to "random" addresses... any addresses generated in an MPMA tx1 are spendable / signable.

**Comment by subterranean1** [comment]
Thank you for the quick response. To be clear, I was not trying to use a Trezor for multisend. I was using my laptop and then desktop.I was directed to a post that mentioned removing hardware wallet addresses from freewallet might fix the problem.I’ll make note of the addresses where the failed tx fees reside in hopes of retrieving them at a later date On Jan 28, 2024, at 11:13 AM, J-Dog ***@***.***> wrote:﻿

I can confirm multisend does not work with Trezor.

@B0BSmiths Correct, I will be adding MPMA support for trezor in the future... just need to update the code a bit so that we handle signing the second tx... Javier and I will dig into this when we have some time.

Yesterday (1/24/24) using Version 0.9.25 AND 0.9.26 I got an error when attempting to do a multi-send.

@subterranean1 FYI, the outputs in MPMA tx1 are still signable/spendable... not random... so, even if you generate part 1 of a 2 part MPMA tx, then part 2 fails... you are still able to gather/collect all the dust from those part1 txs.
Currently no tool exists to find these failed half-way-done MPMA sends and gather the dust up, but its technically possible, and i'm sure tools will be built in the future to provide the ability to gather dust with a few clicks.
TLDR, your not losing any BTC to "random" addresses... any addresses generated in an MPMA tx1 are spendable / signable.
